### ISSUE TITLE
Own image/mask generator as train input?

### ISSUE THREAD

**Issue by Adelost** [divamgupta/image-segmentation-keras#122]
Are there any current way of using a generator when inputing train images and masks instead of using folders? 

E.g. the existing keras.preprocessing.image.ImageDataGenerator that can be used with fit_generator.
https://keras.io/models/sequential/

I would like to use it to create my own image augmentation each epoch without having to save them to disk. 

If not, is it something that you consider adding in the future?

**Comment by divamgupta** [comment]
Yes, the repo uses a generator in the backed. You can create your own generator like that. Pls refer to `data_loader.py` . 

PS: there is a function for augmentation included in the repo.

**Comment by Adelost** [comment]
> Yes, the repo uses a generator in the backed. You can create your own generator like that. Pls refer to `data_loader.py` .
> 
> PS: there is a function for augmentation included in the repo.

Sorry for the late reply. 

Do I have to change the source code in order to use any of the generators, or can they be used in the current state?

It looks like the "data_loader.py" is only used internally and none of the current interfaces takes a generator as an input. The CLI interface seems to have a "do_augment"-flag, but it is only used for visualizing the dataset.
